File: TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/utils/ViewUtils.java
package com.skedgo.tripkit.ui.utils;

import android.text.TextUtils;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.DrawableRes;
import androidx.annotation.NonNull;

public final class ViewUtils {
    private ViewUtils() {
    }

    public static void setText(TextView view, CharSequence text) {
        if (view != null) {
            view.setText(text);
            if (TextUtils.isEmpty(text)) {
                view.setVisibility(View.GONE);
            } else {
                view.setVisibility(View.VISIBLE);
            }
        }
    }

    public static void setImage(@NonNull ImageView view, @DrawableRes int res) {
        if (res != 0) {
            view.setImageResource(res);
            view.setVisibility(View.VISIBLE);
        } else {
            view.setVisibility(View.GONE);
        }
    }
}

File: TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/utils/ViewUtils.kt
package com.skedgo.tripkit.ui.utils

import android.text.TextUtils
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.annotation.DrawableRes

object ViewUtils {
    @JvmStatic
    fun setText(view: TextView?, text: CharSequence?) {
        if (view != null) {
            view.text = text
            if (TextUtils.isEmpty(text)) {
                view.visibility = View.GONE
            } else {
                view.visibility = View.VISIBLE
            }
        }
    }

    fun setImage(view: ImageView, @DrawableRes res: Int) {
        if (res != 0) {
            view.setImageResource(res)
            view.visibility = View.VISIBLE
        } else {
            view.visibility = View.GONE
        }
    }
}